autonick.lua

```diff
@@ -14,41 +14,87 @@ If there are fewer available nicknames than dwarves, the remaining
 dwarves will go un-nicknamed.
 
 You may wish to use this script with the "repeat" command, e.g:
-``repeat -name autonick -time 3 -timeUnits months -command [ autonick ]``
+``repeat -name autonick -time 3 -timeUnits months -command [ autonick -f ]``
 
+Usage:
+
+    autonick [<options>]
+
+Options:
+
+:``-f``, ``--force``:
+    skip confirmation. useful for scripting.
+:``-h``, ``--help``:
+    Show this text.
+:``-q``, ``--quiet``:
+    Do not report how many dwarves were given nicknames.
 ]====]
 
-local seen = {}
---check current nicknames
-for _,unit in ipairs(df.global.world.units.active) do
-    if dfhack.units.isCitizen(unit) and
-    unit.name.nickname ~= "" then
-        seen[unit.name.nickname] = true
-    end
-end
+local options = {}
+
+local argparse = require('argparse')
+local commands = argparse.processArgsGetopt({...}, {
+    {'h', 'help', handler=function() options.help = true end},
+    {'f', 'force', handler=function() options.force = true end},
+    {'q', 'quiet', handler=function() options.quiet = true end},
+})
 
-local names = {}
--- grab list, put in array
-local path = dfhack.getDFPath () .. "/dfhack-config/autonick.txt";
-for line in io.lines(path) do
-    line = line:trim()
-    if (line ~= "")
-    and (not line:startswith('#'))
-    and (not seen[line]) then
-        table.insert(names, line)
-        seen[line] = true
+-- autonick does not use commands, so if there are any, the script is being used wrong.
+if #commands ~= 0 then options.help = true end
+
+if options.help == true then
+    print(dfhack.script_help())
+else
+    result = options.force;
+    if options.force ~= true then
+        local utils = require("utils")
+        result = utils.prompt_yes_no("nickname unnicked dwarves? (use -f to disable this check)", true)
```

Comment:
I'm not sure which is preferable, but an alternative is to display help when there are no arguments and to require something (like the word `all`) to autonick all dwarves. [unforbid](https://docs.dfhack.org/en/stable/docs/_auto/base.html?highlight=unforbid#unforbid) takes this approach, for example. displaying a yes/no prompt seems...heavy handed

Reply:
agreed, that makes much more sense to me.

Reply:
should be good now.